### PR TITLE
BUG: Fix for convert_dtypes with mix of int and string

### LINKED ISSUE #32117: convert_dtypes fails with int and str
#### Code Sample, a copy-pastable example if possible
Thanks for an amazing package and maintaining it

```python
# Your code here
s = pd.Series(["h", "i", 1])
s.convert_dtypes()
```
#### Problem description

Example raises an error, when it shouldn't. I think the case is missing from the tests

**Note**: We receive a lot of issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates!

**Note**: Many problems can be resolved by simply upgrading `pandas` to the latest version. Before submitting, please check if that solution works for you. If possible, you may want to check if `master` addresses this issue, but that is not necessary.

For documentation-related issues, you can check the latest versions of the docs on `master` here:

https://pandas-docs.github.io/pandas-docs-travis/

If the issue has not been resolved there, go ahead and file it in the issue tracker.

#### Expected Output
series with object dtype ?

#### Output of ``pd.show_versions()``

[paste the output of ``pd.show_versions()`` here below this line]
In [6]: pd.show_versions()                                                                                                                 

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.2.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.2.0
Cython           : None
pytest           : 5.3.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pytest           : 5.3.5
pyxlsb           : None
s3fs             : 0.4.0
scipy            : 1.4.1
sqlalchemy       : 1.3.13
tables           : None
tabulate         : 0.8.6
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None
<details>

</details>

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] closes #32117
- [x] tests added / passed
  - added new cases for `test_convert_dtypes`
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry
  - placed in v1.0.2 whatsnew

### CODE REVIEW

File: pandas/tests/series/methods/test_convert_dtypes.py [hunk 1]
```diff
@@ -144,11 +156,23 @@ class TestSeriesConvertDtypes:
                 [1, 2.0],
```
Comment (jorisvandenbossche) [inline pandas/tests/series/methods/test_convert_dtypes.py]: It's a bit hard to interpret below tests / diff, but does pd.Series([1, 2.0], dtype=object).convert_dtypes() still give Int64?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! Small question

File: pandas/tests/series/methods/test_convert_dtypes.py (hunk 1 above)
Comment (Dr-Irv) [reply pandas/tests/series/methods/test_convert_dtypes.py]: Yes.  Here's the interpretation of the tests for that case:
The code reads:
```python
            (
                [1, 2.0],
                object,
                {
                    ((True,), (True, False), (True,), (True, False)): "Int64",
                    ((True,), (True, False), (False,), (True, False)): np.dtype(
                        "float"
                    ),
                    ((False,), (True, False), (True, False), (True, False)): np.dtype(
                        "object"
                    ),
                },
            ),
```
This means the following:
1) Create a `Series` with `[1, 2.0]` as the entries, with dtype `object`
2) Consider the 16 possible combinations of the 4 arguments `infer_objects`, `convert_string`, `convert_integer` and `convert_boolean`
3) If `infer_objects==True` and `convert_integer==True`, result should be `Int64`
4) If `infer_objects==True` and `convert_integer==False`, result should be `float`
5) If `infer_objects==False`, result is always `object`

Prior to this PR, the tests were as follows:
p3) If `convert_integer==True`, result should be `Int64` independent of value of `infer_objects`
p4) If `infer_objects==True` and `convert_integer==False`, result should be `float` (same)
p5) If `infer_objects==False` and `convert_integer==False`, result is `object`

I think the new version is what we want the behavior to be, i.e., if you start with `object` and  you don't do the infer-objects step, it remains an object.

Comment (jorisvandenbossche) [reply pandas/tests/series/methods/test_convert_dtypes.py]: OK, thanks

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@Dr-Irv Thanks!

**Pull request merged**